tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/object/test_bir_positive.bal

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import bir/objs;
+
+objs:Qux quxMod = isolated service object {
+    public isolated function execute() returns anydata|error {
+         return "qux";
+    }
+};
+
+function f1(objs:Foo foo) {
+    assertTrue(foo is objs:Foo);
+}
+
+function f2(objs:Xyz xyz) {
+    assertTrue(xyz is objs:Xyz);
+}
+
+function f3(any foo) {
+    assertTrue(foo is objs:Foo);
+}
+
+function f4(any baz) {
+    assertTrue(baz is objs:Baz);
+}
+
+function f5(any xyz) {
+    assertTrue(xyz is objs:Xyz);
+}
+
+function f6(any qux) {
+    assertTrue(qux is objs:Qux);
+}
+
+function f7(any foo) {
+    assertFalse(foo is objs:Bar);
+}
+
+function testObjectTypeAssignability() {
+    objs:Foo foo = isolated client object {
+         public isolated function execute() returns anydata|error {
+           return "foo";
+        }
+    };
+    f1(foo);
```

Comment:
Do we really need the separate functions? Can't we do the checks here itself? That would help with readability IMO, since we won't have to go to and from functions.

Reply:
These functions were added to address this [#suggestion](https://github.com/ballerina-platform/ballerina-lang/pull/40901#discussion_r1257754638). By removing the functions we cannot achieve the assigning validation.

Reply:
Why can't we do

```suggestion
    assertTrue(<any> foo is objs:Foo);
```

or 

```suggestion
    any a = foo;
    assertTrue(a is objs:Foo);
```